core/arch/arm/arm.mk

```diff
@@ -308,8 +321,96 @@ ta-mk-file-export-add-ta_arm64 += COMPILER_ta_arm64 ?= $$(COMPILER)_nl_
 ta-mk-file-export-add-ta_arm64 += PYTHON3 ?= python3_nl_
 endif
 
-# Set cross compiler prefix for each TA target
-$(foreach sm, $(ta-targets), $(eval CROSS_COMPILE_$(sm) ?= $(CROSS_COMPILE$(arch-bits-$(sm)))))
+ifneq ($(filter sp_arm32,$(sp-targets)),)
+# Variables for sp-target/sm "sp_arm32"
+CFG_ARM32_sp_arm32 := y
+arch-bits-sp_arm32 := 32
+sp_arm32-platform-cppflags += $(arm32-platform-cppflags)
+sp_arm32-platform-cflags += $(arm32-platform-cflags)
+sp_arm32-platform-cflags += $(platform-cflags-optimization)
+sp_arm32-platform-cflags += $(platform-cflags-debug-info)
+sp_arm32-platform-cflags += -fpic
+
+# Thumb mode doesn't support function graph tracing due to missing
+# frame pointer support required to trace function call chain. So
+# rather compile in ARM mode if function tracing is enabled.
+ifeq ($(CFG_FTRACE_SUPPORT),y)
```

Comment:
ftrace (from libutils) in TAs currently mandates some specific syscall support from libutee.
Maybe somewhere it should be stated that `CFG_FTRACE_SUPPORT` must be disabled when generating the SP dev kit.

Reply:
I've added a check and an error message for SP dev kit build & `CFG_FTRACE_SUPPORT=y` cases.